procbuild/builder.py

```diff
@@ -80,7 +80,172 @@ def checkout(repo, branch, build_path):
     return shell('git clone %s --branch %s --single-branch %s' %
                  (repo, branch, build_path), retry=4)
 
+class BuildError(Exception):
+    
+    def __init__(self, message):
+        self.message = f"A build error occurred during the self.{message} step."
+
+class BuildManager(object):
+    
+    def __init__(self, 
+                 user, 
+                 branch, 
+                 target, 
+                 master_branch='master', 
+                 log=None):
+        self.user = user
+        self.build_output = ''
+        self.status = 'fail'
+        self.branch = branch
+        self.target = target
+        self.master_branch = master_branch
+        self.build_status = 'Build started...'
+        self.build_pdf_path = ''
+        self.master_repo_path = joinp(cache(), 'scipy_proceedings')
+        self.build_timestamp = time.strftime('%d/%m %H:%M')
+        self.target_path = joinp(cache(), '%s.pdf' % target)
```

Comment:
Should this be an input parameter as well?

Reply:
yes